### ISSUE TITLE
New Feature : Possibility to lock screen rotation

### ISSUE THREAD

**Issue by djelloul78** [brarcher/loyalty-card-locker#114]
I'd like to know if it's possible to add a button or something on the barcode screen to lock the screen orientation?
It is sometimes useful to avoid the screen to move when we present the phone to the barcode scanner or the cashier.
Especially when you have a cart completely full, with a lot of people looking at you, because they commonly decided they are late because of you...

**Comment by Entze** [comment]
That's usually the OSes domain isn't it?

**Comment by djelloul78** [comment]
Yes it is. But it needs to access the phone settings or create a specific shortcut outside the application.
For a better application usage, I think it's better to add this possibility directly into the app for an increased experience.
But you're probably right, this feature request is maybe not so useful as I think...
We'll let the author of the app and the other users give their feedback about it.

**Comment by brarcher** [comment]
I believe that there is a way to restrict an application from responding to rotation, or forcing it to be landscape or portrait. It would be fairly simple to allow one to click on the barcode, then open it into a new screen that is landscape only. Is that was you are looking for?

**Comment by djelloul78** [comment]
I didn't think about this clever idea, but yes, your proposal is a good one.
Do you think that it could be possible to make the barcode switching between landscape and portrait mode each time we touch it in the newly opened screen you suggested?
Excuse me if I insist on this, but I have a specific case in mind...
In fact, I'm used to make drive-in shopping regularly.
In this situation, you have to use a completely automated retreival/payement terminal.
In front of this device, you can enter your ordering number, your fidelity card number or you can scan your fidelity card barcode.
The slot is able to let us enter a phone for scanning, but only by the smallest width.
As you probably guessed, if the barcode is not properly oriented, there's no chance for it to be read by the terminal... Requesting me to search for my card to be scanned!
Sorry for the long explanation story.

**Comment by brarcher** [comment]
After hearing your use case and thinking about this more, I think a good solution is to add a lock icon in the menu bar. Here is an example I see from another application:

![image](https://user-images.githubusercontent.com/5264535/28148814-19eba06c-6757-11e7-838d-f39cfdb54315.png)

If locked the screen rotation would be disabled, and unlocked would allow rotation.

I've passed my phone to someone at a store to scan, and I've observed sometimes the phone orientation would change, causing some frustration.

I'll look into adding this feature. Thanks for the suggestion.
